src/components/DefaultModal.tsx

```diff
@@ -0,0 +1,116 @@
+import React from 'react';
+import styled from 'styled-components';
+import MainButton, { Variant } from './MainActionButton';
+
+export const Overlay = styled.div`
+	position: absolute;
+	backdrop-filter: blur(5px);
+	height: 100%;
+	width: 100%;
+	display: flex;
+	// justify-content: center;
+	align-items: center;
+`;
+
+export const ButtonField = styled.div`
+	margin-right: 2em;
+	margin-left: 7em;
+	width: 100%;
+	display: grid;
+	grid-template-columns: 1fr 1fr;
+	grid-template-areas: 'left right';
+	grid-gap: 1em;
+`;
+
+const ModalContainer = styled.div`
+	background-color: #ffffff;
+	position: fixed;
+	display: grid;
+	grid-template-rows: auto 1fr auto;
+	grid-template-areas:
+		'header'
+		'body'
+		'footer';
+	border-radius: 30px;
+	overflow: auto;
+	width: 24em;
+	height: auto;
+	&:not {
+		background-color: red;
+	}
+`;
+
+const Header = styled.div`
+	background-color: #777777;
+	display: flex;
+	width: 100%;
+	height: 3em;
+	justify-content: center;
+	align-items: center;
+	grid-area: header;
+`;
+
+// Footer
+const Footer = styled.div`
+	display: flex;
+	align-items: center;
+	background-color: #f8f8f8;
+	width: 100%;
+	height: 3.5em;
+	grid-area: footer;
+`;
+
+export interface ModalInterface {
+	color?: string;
+	children: React.ReactNode;
+	header: string | React.ReactNode;
+	primaryButton: string;
+	primaryVar: Variant;
+	secondaryVar: Variant;
+	secondaryButton: string;
+	handleClose: () => void;
+	handlePrimary: () => void;
+}
+
+function DefaultModal({
+	color,
+	header,
+	children,
+	primaryButton,
+	primaryVar,
+	secondaryButton,
+	secondaryVar,
+	handleClose,
+	handlePrimary,
+}: ModalInterface) {
+	return (
+		<Overlay onClick={() => handleClose()}>
+			<ModalContainer
+				onClick={(e) => {
+					e.stopPropagation();
```

Comment:
Yeah! We wanted the modal to close whenever the user clicks outside the modal, hence the `onClick` function property on `<Overlay>`; however, since the click function is on the parent element of `<ModalContainer>` whenever you click on the Modal Container the event will bubble up (event propagation) to the parent element, thus closing the modal.

React actually has an event method that stops the event (the click) from also being sent to `Overlay`. All that's happening is whenever `ModalContainer` is clicked by the user, that event (`e`) is caught by the `stopPropagation()` and the event is no longer sent to `Overlay`, thus not closing the modal.

tl;dr - `stopPropagation()` prevents the modal from closing whenever the user clicks inside it :)

[Here's an article explaining bubbling and propagation in further detail](https://www.sitepoint.com/event-bubbling-javascript/)

Reply:
Awesome that makes sense, thanks!! :)